### DISCUSSION TITLE
Using custom gcc version

### DISCUSSION THREAD

**Discussion by mahmoodn** [apache/incubator-mxnet#16121] [Q&A] [answered]
Is there any way to use custom gcc version instead of the default on the system? A method other than exporting CC, CXX.

**Comment by zachgk** [comment]
Hi @mahmoodn. If you are using the MAKE base build instead of the CMake one, then you can pass in a new value for CXX through `make CXX=xxx ...`. You can also use bash right before the command such as `CXX=xxx someBashCommand...`. 

Can you explain what your use case is, how you are building mxnet, and what prevents you from passing in the gcc version through the CC and CXX environment variables?

**Comment by mahmoodn** [comment]
I had a compilation error using gcc-4.8.5 on centos 7.
I can reproduce the error but I think it was related to older gcc version.
So, I decided to test a newer gcc, e.g. 7.

Is there any note about minimum version of GCC for the latest git version?

**Comment by leezu** [comment, marked as answer]
Officially currently MXNet only supports gcc-4.8 build: https://github.com/apache/incubator-mxnet/blob/681cfc4/tools/dependencies/README.md

Can you provide more details on the build issues with gcc-4.8.5?
